docs/reference/remap.cue

```diff
@@ -65,4 +64,205 @@ remap: {
 			description: "Raised when the provided input cannot be parsed."
 		}
 	}
+
+	// TRL type system
+	types: [TypeName=string]: {
+		#Use: "parameter" | "return"
+
+		description: string
+		use: [#Use, ...#Use]
+	}
+
+	types: {
+		"any": {
+			description: "A stand-in for any type."
+			use: ["parameter"]
+		}
+		"array": {
+			description: "A list of items."
+			use: ["parameter", "return"]
```

Comment:
Should `assignment` also be in this list? On second thought, I'm curious if we even need this list?

Reply:
Given that there are subtle differences in the two lists, I do think that it would be good to at least register this information here. I tried to scheme a way to derive it from the enums above instead of duplicating but I'm not sure CUE provides a path there.